include/ensmallen_bits/lbfgs/lbfgs_impl.hpp

```diff
@@ -212,6 +224,13 @@ bool L_BFGS::LineSearch(FunctionType& function,
   double stepSize = 1.0;
   finalStepSize = 0.0; // Set only when we take the step.
 
+  if (searchDirection.is_finite() == false)
+  {
+    Warn << "L-BFGS line search direction vector is not finite "
+        << "(terminating)!" << std::endl;
+    return false;
+  }
```

Comment:
Personally I think that the better check would be on `initialSearchDirectionDotGradient`; if `searchDirection` isn't finite, then that won't be either.  And if `gradient` isn't finite, the same thing will happen, so checking the scalar `initialSearchDirectionDotGradient` seems like it is basically the same level of error detection for much cheaper.

Reply:
good point; implemented in https://github.com/mlpack/ensmallen/pull/368/commits/49ad90c6193dfdd86c9be7b155473ee738d07685